### ISSUE TITLE
Collapse and show children toggle

### ISSUE THREAD

**Issue by BentleyDavis** [ReasonScore/reason-score-react#5]
Have an triangle/arrow  just outside and to the left of the claim. Click to open and close.
When closed it will point to the right.
When open it points down.
Animate rotation

**Comment by BentleyDavis** [comment]
Can you put this into a pull request so I can review it easily?

**Comment by reut-soficoop** [comment]
I guess I didn't push my commits. you can check it now. I will put in pull request from now on.

**Comment by reut-soficoop** [comment]
no. I saw this comment after I already pushed. so from now on...

**Comment by BentleyDavis** [comment]
I just looked at the code and it appears you did not read carefully enough. 

- In the title I use the term "Children" which in the code refers to child Claims. You made it so that the the triangle rotates when the Editor opens and closes, not the children.
- to be clearer, I should have said "Have an triangle/arrow just outside and to the left of the claim. Click the **arrow/triangle** to open and close **the children**."  
- Submit a pull request with these changes. Prioritize this over working on the Firestore.

**Comment by BentleyDavis** [comment]
Please animate the showing and hiding of the children as well.

**Comment by BentleyDavis** [comment]
I do not understand what you are saying. In the app right now there are three debates. The top two debates have children. When I click on the arrow the arrow rotates down slowly but there child claim abruptly appears without any animation. It should expand vertically slowly.

**Comment by BentleyDavis** [comment]
It there are no children the arrow should not be visible

**Comment by avirankatzsofi** [comment]
Children toggle animation was added in 3cdc3d9.
Please note that this is still buggy and behaves strangely when there are nested claims.

**Comment by BentleyDavis** [comment]
@avirankatzsofi Please check out https://github.com/ReasonScore/reason-score-react/tree/feature/%235-collapse-and-show-children-toggle for my suggestion on how to manage local state for individual claims/arguments that does not need to be persisted to the database. As you said, we will probably need to completely re-design how buildViewModel works so that the system does not re-draw everything with every update. Let me know what you think about my change for now.

**Comment by avirankatzsofi** [comment]
I merged your solution to master
